### PR TITLE
Updates to Java branching doc (NR-56333)

### PR DESCRIPTION
**Pull request by paperclypse**
Making some tweaks and improvements.

### CODE REVIEW

File: src/install/java/license.mdx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (bradleycamacho) [inline src/install/java/license.mdx:1]: I'm not opposed either way, but why not follow the normal pattern of linking out to the license key doc here?

**Review by bradleycamacho — APPROVED** [review]
approved except one small comment

File: src/install/java/license.mdx (hunk 1 above)
Comment (paperclypse) [reply src/install/java/license.mdx:1]: Good question! The main reason is that I think our license key doc sort of muddies the waters a bit. And also I'd rather someone just went and did the thing than read a doc about it. But if you think it's worth linking to, I'm happy to.

Comment (austin-schaefer) [reply src/install/java/license.mdx:1]: Generally links into the UI are a good conversion spot, so I think I'd prefer a direct link over a link to the (somewhat complicated) license key doc

**Comment by LizBaker** [comment]
re-trigger gatsby

File: src/install/java/license.mdx (hunk 1 above)
Comment (paperclypse) [reply src/install/java/license.mdx:1]: I've come up with a both/and solution here. We can always remove the link to the license key doc later if we want to.

**Pull request merged**